graphql/src/lib.rs

```diff
@@ -154,3 +103,42 @@ pub fn run_server(server_data: ServerData, addr: SocketAddr) -> Result<Server> {
     });
     Ok(server.bind(addr)?.run())
 }
+
+struct SessionAndHandler {
+    pub session: u32,
+    pub handler: Arc<dyn HandleGraphQlRequest>,
+}
+
+impl FromRequest for SessionAndHandler {
+    type Error = Error;
+    type Future = Pin<Box<dyn Future<Output = Result<Self, Self::Error>>>>;
```

Comment:
I don't know well about this type `Pin<Box<dyn Future`. I found an example and followed.

Reply:
https://stackoverflow.com/questions/63308246/how-to-use-async-code-in-actix-web-extractors

Reply:
AFAIK returning associated Future object is workaround of async trait. http://smallcultfollowing.com/babysteps/blog/2019/10/26/async-fn-in-traits-are-hard/

Reply:
Thanks!